packages/table/src/common/requestIdleCallback.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+/**
+ * Event name for `postMessage`
+ */
+const MESSAGE_EVENT_DATA = "blueprint-table-post-message";
+
+type Callback = () => void;
+
+/**
+ * Object that holds state for managing idle callbacks
+ */
+const IDLE_STATE = {
+    callbacks: [] as Callback[],
+    triggered: false,
+};
+
+const handleIdle = (event: MessageEvent) => {
+    if (event.source !== window || event.data !== MESSAGE_EVENT_DATA) {
+       return;
+    }
+
+    IDLE_STATE.triggered = false;
+
+    let callback = null;
+    if (IDLE_STATE.callbacks.length > 0) {
+        callback = IDLE_STATE.callbacks.shift();
+    }
+
+    if (IDLE_STATE.callbacks.length > 0) {
+        triggerIdleFrame();
+    }
+
+    // finally, invoke the callback. exceptions will be propagated
+    if (callback) {
```

Comment:
Can we use `Utils.safeInvoke` from `@blueprintjs/core`? May not be worth the cross-package dependency though.